node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/acl/AclMaintainer.java

```
@@ -1,99 +1,113 @@
- * containers (e.g. any container, except node-admin).
+ * The configuration will be retried the next time the maintainer runs.
  *
  * @author mpolden
+ * @author smorgrav
  */
 public class AclMaintainer implements Runnable {
 
     private static final PrefixLogger log = PrefixLogger.getNodeAdminLogger(AclMaintainer.class);
-    private static final String IPTABLES_COMMAND = "ip6tables";
 
     private final DockerOperations dockerOperations;
     private final NodeRepository nodeRepository;
+    private final IPAddresses ipAddresses;
     private final String nodeAdminHostname;
-    private final Map<ContainerName, Acl> containerAcls;
 
-    public AclMaintainer(DockerOperations dockerOperations, NodeRepository nodeRepository,
-                         String nodeAdminHostname) {
+    public AclMaintainer(DockerOperations dockerOperations, NodeRepository nodeRepository, String nodeAdminHostname, IPAddresses ipAddresses) {
         this.dockerOperations = dockerOperations;
         this.nodeRepository = nodeRepository;
+        this.ipAddresses = ipAddresses;
         this.nodeAdminHostname = nodeAdminHostname;
-        this.containerAcls = new HashMap<>();
     }
 
-    private boolean isAclActive(ContainerName containerName, Acl acl) {
-        return Optional.ofNullable(containerAcls.get(containerName))
-                .map(acl::equals)
-                .orElse(false);
-    }
+    private void apply(Container container, Acl acl, IPVersion ipVersion) {
 
-    private void applyAcl(ContainerName containerName, Acl acl) {
-        if (isAclActive(containerName, acl)) {
-            return;
-        }
-        final Command flush = new FlushCommand(Chain.INPUT);
-        final Command rollback = new PolicyCommand(Chain.INPUT, Action.ACCEPT);
+        // 1. Get address
+        Optional<InetAddress> address = ipAddresses.getAddress(container.hostname, ipVersion);
```

This should probably throw an exception and the caller needs to know which IP addresses this host should have.

---

The normal case is where does not return an address is when the container does not have a ipv4 address (at least not DNS wise) which is the case for our tenant containers today. We could move that logic to the caller of course.